stm32/Src/main.c

```diff
@@ -84,8 +109,116 @@ int main(void)
 
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
+  MX_USART1_UART_Init();
+  MX_I2C2_Init();
   /* USER CODE BEGIN 2 */
-  MX_GPIO_Init();
+
+  //uint32_t value_adc;
+  char output[100];

```

Comment:
Move all test code to independent test folder, see [this](https://docs.platformio.org/en/stable/advanced/unit-testing/index.html)